src/client/java/teammates/client/scripts/sql/VerifyCourseEntityAttributes.java

```diff
@@ -154,14 +163,180 @@ private boolean verifyStudent(CourseStudent oldStudent,
     }
 
     private List<Student> getNewStudents(String courseId) {
-        HibernateUtil.beginTransaction();
+        // HibernateUtil.beginTransaction();
         CriteriaBuilder cb = HibernateUtil.getCriteriaBuilder();
         CriteriaQuery<teammates.storage.sqlentity.Student> cr = cb
                 .createQuery(teammates.storage.sqlentity.Student.class);
         Root<teammates.storage.sqlentity.Student> courseRoot = cr.from(teammates.storage.sqlentity.Student.class);
         cr.select(courseRoot).where(cb.equal(courseRoot.get("courseId"), courseId));
         List<Student> newStudents = HibernateUtil.createQuery(cr).getResultList();
-        HibernateUtil.commitTransaction();
+        // HibernateUtil.commitTransaction();
         return newStudents;
     }
+
+    // methods for verify feedback chain -----------------------------------------------------------------------------------
+
+    private boolean verifyFeedbackChain(teammates.storage.sqlentity.Course newCourse) {
+        List<teammates.storage.sqlentity.FeedbackSession> newSessions = newCourse.getFeedbackSessions();
+        List<FeedbackSession> oldSessions = ofy().load().type(FeedbackSession.class)
+                .filter("courseId", newCourse.getId()).list();
+
+        if (newSessions.size() != oldSessions.size()) {
+            log(String.format("Mismatched session counts for course id: %s", newCourse.getId()));
+            return false;
+        }
+
+        Map<String, FeedbackSession> sessionNameToOldSessionMap = oldSessions.stream()
+                .collect(Collectors.toMap(FeedbackSession::getFeedbackSessionName, session -> session));
+
+        return newSessions.stream().allMatch(newSession -> {
+            FeedbackSession oldSession = sessionNameToOldSessionMap.get(newSession.getName());
+            return verifyFeedbackSession(oldSession, newSession);
+        });
+    }
+
+    private boolean verifyFeedbackSession(FeedbackSession oldSession, teammates.storage.sqlentity.FeedbackSession newSession) {
+        boolean doFieldsMatch = newSession.getCourse().getId().equals(oldSession.getCourseId())
+                && newSession.getName().equals(oldSession.getFeedbackSessionName())
+                && newSession.getCreatorEmail().equals(oldSession.getCreatorEmail())
+                && newSession.getInstructions().equals(SanitizationHelper.sanitizeForRichText(oldSession.getInstructions()))
```

Comment:
Is `newSession.getInstructions` automatically sanitised? Could the sanitiser change the name causing this to fail?

Reply:
I'm not sure about this, as I copied it directly from `VerifyFeedbackSessionAttributes`. 

@yuanxi1, was there some reason for this?

Reply:
Shall we just sanitize during migration and we can leave this line in?

Reply:
When I tested it one entity in the typical data bundle (which tests for sanitizing injection in the instruction field) has one more space character than the saved sql version, and the reason is that sanitization I believe is forced before instructions are saved (sql), and it will strip more stuff than what datastore does.